src/common/commonutils/UserUtils.c

```diff
@@ -819,49 +816,37 @@ int CheckNoDuplicateUidsExist(char** reason, OsConfigLogHandle log)
     return status;
 }
 
-int RemoveUser(SimplifiedUser* user, bool removeHome, OsConfigLogHandle log)
+static int RemoveUser(const char* username, const uid_t uid, const gid_t gid, const char* home, OsConfigLogHandle log)
 {
-    const char* commandTemplate = "userdel %s %s";
-    char* command = NULL;
-    int status = 0, _status = 0;
+    const char* commandTemplate = "userdel -f %s";
+    char command[64] = {0};
+    int status = 0;
+    int commandLen = 0;
 
-    if (NULL == user)
+    if (NULL == username)
     {
         OsConfigLogError(log, "RemoveUser: invalid argument");
         return EINVAL;
     }
-    else if (0 == user->userId)
+
+    commandLen = snprintf(command, sizeof(command), commandTemplate, username);
+    if(((int)sizeof(command) <= commandLen) || (0 > commandLen))
```

Comment:
This PR says it simplifies the code. But here actually there is an added complication. Instead of using a single call, FormatAllocateString to allocate and initialize the command string, we have this static buffer, then length calculation, then snprintf call, then the error check etc?

The reason for having this function, FormatAllocateString, is exactly for cases like this. Let's use it more.